blaze/expr/expressions.py

```diff
@@ -790,6 +804,92 @@ def coerce(expr, to):
     return Coerce(expr, dshape(to) if isinstance(to, _strtypes) else to)
 
 
+class Cast(Expr):
```

Comment:
We'll now have both `cast` and `coerce` expressions with this.  Is the essential that `cast` is for the expression's dshape, while `coerce` is for the underlying backend?